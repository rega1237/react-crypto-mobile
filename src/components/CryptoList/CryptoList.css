.filter-container {
  display: flex;
  justify-content: center;
}

#filter {
  margin: 0 0.5rem;
  padding: 0.5rem;
}

.all-coins {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(150px, 1fr));
  padding-top: 15px;
}
