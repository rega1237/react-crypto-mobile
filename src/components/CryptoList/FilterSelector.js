import PropTypes from 'prop-types';

const FilterSelector = (props) => {
  const { selectHandler } = props;

  return (
    <select name="filter" id="filter" onInput={(e) => selectHandler(e)}>
      <option defaultValue="All">All</option>
      <option value="10">Rank 1-10</option>
      <option value="20">Rank 11-20</option>
      <option value="30">Rank 21-30</option>
      <option value="40">Rank 31-40</option>
      <option value="50">Rank 41-50</option>
      <option value="60">Rank 51-60</option>
      <option value="70">Rank 61-70</option>
      <option value="80">Rank 71-80</option>
      <option value="90">Rank 81-90</option>
      <option value="100">Rank 91-100</option>
    </select>
  );
};

FilterSelector.propTypes = {
  selectHandler: PropTypes.func.isRequired,
};

export default FilterSelector;
