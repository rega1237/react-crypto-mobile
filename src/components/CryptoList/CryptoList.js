import { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchCoins } from '../../redux/reducer';
import Coin from '../Coin/Coin';
import FilterSelector from './FilterSelector';
import './CryptoList.css';

const CryptoList = () => {
  const cryptoStore = useSelector((state) => state.cryptoReducer);
  const dispatch = useDispatch();
  const [filterCoin, setFilter] = useState([]);

  if (cryptoStore.length === 0) {
    dispatch(fetchCoins);
  }

  const selectHandler = (e) => {
    if (e.target.value === '10') {
      setFilter(cryptoStore.filter((crypto) => crypto.rank <= 10));
    } else if (e.target.value === '20') {
      setFilter(cryptoStore.filter((crypto) => crypto.rank <= 20 && crypto.rank >= 11));
    } else if (e.target.value === '30') {
      setFilter(cryptoStore.filter((crypto) => crypto.rank <= 30 && crypto.rank >= 21));
    } else if (e.target.value === '40') {
      setFilter(cryptoStore.filter((crypto) => crypto.rank <= 40 && crypto.rank >= 31));
    } else if (e.target.value === '50') {
      setFilter(cryptoStore.filter((crypto) => crypto.rank <= 50 && crypto.rank >= 41));
    } else if (e.target.value === '60') {
      setFilter(cryptoStore.filter((crypto) => crypto.rank <= 60 && crypto.rank >= 51));
    } else if (e.target.value === '70') {
      setFilter(cryptoStore.filter((crypto) => crypto.rank <= 70 && crypto.rank >= 61));
    } else if (e.target.value === '80') {
      setFilter(cryptoStore.filter((crypto) => crypto.rank <= 80 && crypto.rank >= 71));
    } else if (e.target.value === '90') {
      setFilter(cryptoStore.filter((crypto) => crypto.rank <= 90 && crypto.rank >= 81));
    } else if (e.target.value === '100') {
      setFilter(cryptoStore.filter((crypto) => crypto.rank <= 100 && crypto.rank >= 91));
    } else {
      setFilter(cryptoStore);
    }
  };

  return (
    <div className="all-coins-container">
      <div className="filter-container">
        <FilterSelector selectHandler={selectHandler} />
      </div>
      <div className="all-coins">
        {filterCoin.length > 0
        && filterCoin.map((crypto) => (
          <Coin
            key={crypto.rank}
            img={crypto.img}
            id={crypto.id}
            name={crypto.name}
            rank={crypto.rank}
            price={crypto.priceUsd}
            volume={crypto.volumeUsd24Hr}
          />
        ))}
        {filterCoin.length === 0
        && cryptoStore.map((crypto) => (
          <Coin
            key={crypto.rank}
            img={crypto.img}
            id={crypto.id}
            name={crypto.name}
            rank={crypto.rank}
            price={crypto.priceUsd}
            volume={crypto.volumeUsd24Hr}
          />
        ))}
      </div>
    </div>
  );
};

export default CryptoList;
