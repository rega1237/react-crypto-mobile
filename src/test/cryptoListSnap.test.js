import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import CryptoList from '../components/CryptoList/CryptoList';
import store from '../redux/configureStore';

test('CryptoList is rendering', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <CryptoList />
        </MemoryRouter>
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
