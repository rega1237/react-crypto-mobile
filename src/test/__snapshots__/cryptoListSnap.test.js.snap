// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoList is rendering 1`] = `
<div
  className="all-coins-container"
>
  <div
    className="filter-container"
  >
    <select
      id="filter"
      name="filter"
      onInput={[Function]}
    >
      <option
        defaultValue="All"
      >
        All
      </option>
      <option
        value="10"
      >
        Rank 1-10
      </option>
      <option
        value="20"
      >
        Rank 11-20
      </option>
      <option
        value="30"
      >
        Rank 21-30
      </option>
      <option
        value="40"
      >
        Rank 31-40
      </option>
      <option
        value="50"
      >
        Rank 41-50
      </option>
      <option
        value="60"
      >
        Rank 51-60
      </option>
      <option
        value="70"
      >
        Rank 61-70
      </option>
      <option
        value="80"
      >
        Rank 71-80
      </option>
      <option
        value="90"
      >
        Rank 81-90
      </option>
      <option
        value="100"
      >
        Rank 91-100
      </option>
    </select>
  </div>
  <div
    className="all-coins"
  />
</div>
`;
